
PACKAGE ='Base_Action'

from dynamic_reconfigure.parameter_generator_catkin import *
# 2.创建生成器
gen = ParameterGenerator()

# 3.向生成器添加若干参数 action 的全局参数
#add(name, paramtype, level, description, default=None, min=None, max=None, edit_method="")

# many_enum = gen.enum([gen.const("small",int_t,0,"a small size"),
#                 gen.const("mediun",int_t,1,"a medium size"),
#                 gen.const("big",int_t,2,"a big size")
#                 ],"a car size set")

# gen.add("list_param",int_t,0,"列表参数",0,0,2, edit_method=many_enum)

# 重置默认configure  默认关
gen.add("restore_defaults_", bool_t, 0, "Restore to the original configuration", False)
# 重复测试action    默认关
gen.add("repeatTest_",bool_t,0," repeat  Test the action or not ",False)
# 机器臂动作库测试类型 默认关 不开启 默认充电
gen.add("armtask_type_",str_t,0,"arm test action type  default is close ","charge")
# 机械臂是动作库是否开启 默认关
gen.add("action_open_",bool_t,0,"action is open or not ",False)
# 机器臂速度范围设定
gen.add("int_param",int_t,0," int ",50,0,100)
gen.add("double_param",double_t,0,"double param",1.57,0,3.14)

# 4.生成中间文件并退出
exit(gen.generate(PACKAGE,"base_action_node","BaseAction"))